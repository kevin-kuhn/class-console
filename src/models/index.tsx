// PRIMARY RESPONSE
export interface IMeta {
  title: string
  description: string
}

export interface IPrimaryChildren {
  type: string
  title: string
  slug: string
}

export interface IResult {
  title: string
  type: string
  slug: string
  description: string
  children: IPrimaryChildren[] | IContentChildren[]
}

// CONTENT RESPONSE
export interface IVideo {
  provider: string
  link: string
  durationInMinutes: number
}

export interface IPdf {
  title: string
  description: string
  link: string
}

export interface IText {
  title: string
  html: string
}

export interface IAnswer {
  text: string
  isCorrect: boolean
}

export interface IExercice {
  question: string
  answers: IAnswer[]
  correction: string
}

export interface IContentChildren {
  type: string
  data: IVideo | IText | IExercice
}

// FULL RESPONSE
export interface IResponse {
	meta: IMeta
	result: IResult
}