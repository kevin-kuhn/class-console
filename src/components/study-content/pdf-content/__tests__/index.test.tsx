import { render } from '../../../../test/test-utils'
import PdfContent from '../index'

import { IPdf } from '../../../../models'

import '@testing-library/jest-dom'

describe('PdfContent', () => {
  it('should be in document', () => {
    const { container } = render(
      <PdfContent result={{} as IPdf} handleOnDoneStudy={() => {}} />
    )

    expect(container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(
      <PdfContent result={{} as IPdf} handleOnDoneStudy={() => {}} />
    )

    expect(container).toMatchSnapshot()
  })
})
