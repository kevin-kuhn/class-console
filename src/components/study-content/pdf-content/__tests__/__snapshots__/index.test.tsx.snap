// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfContent should match snapshot 1`] = `
<div>
  <a
    class="a"
    target="_blank"
  >
    Acesse o conte√∫do
  </a>
</div>
`;
